### PR TITLE
Remove langversion restriction for source generators.

### PR DESCRIPTION
**Pull request by chsienki**
Closes https://github.com/dotnet/roslyn/issues/42565

### CODE REVIEW

**Comment by jcouv** [comment]
No test was affected?

**Comment by chsienki** [comment]
We didn't have any tests verifying it AFAICT. I can add one that checks it works with lesser lang ver now tho.

File: src/Workspaces/CSharp/Portable/Workspace/LanguageServices/CSharpCompilationFactoryService.cs [hunk 1]
```diff
@@ -45,11 +45,6 @@ CompilationOptions ICompilationFactoryService.GetDefaultCompilationOptions()
 
         GeneratorDriver? ICompilationFactoryService.CreateGeneratorDriver(ParseOptions parseOptions, ImmutableArray<ISourceGenerator> generators, AnalyzerConfigOptionsProvider optionsProvider, ImmutableArray<AdditionalText> additionalTexts)
         {
-            // https://github.com/dotnet/roslyn/issues/42565: for now we gate behind langver == preview. We'll remove this before final shipping, as the feature is langver agnostic
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Workspace/LanguageServices/CSharpCompilationFactoryService.cs:48]: Can you also update this line?

https://github.com/dotnet/roslyn/blob/9737cff23a0064b6b070bee3d4bc356ff49925ae/src/Workspaces/CoreTest/SolutionTests/SolutionWithSourceGeneratorTests.cs#L37

(just delete the parseOptions parameter entirely)

**Review by jasonmalinowski — APPROVED** [review]
:shipit:, but with an appropriate update to the tests first. :smile:

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by davidwengier** [comment]
🎉

**Review by jaredpar — APPROVED** [review]

**Pull request merged**